src/main/java/com/unstoppabledomains/resolution/contracts/uns/ProxyReader.java

```diff
@@ -95,6 +95,12 @@ public List<Location.LocationBuilder> getLocationAddresses(BigInteger... tokenID
         return locations;
     }
 
+    public BigInteger getReverseResolution(String address) throws NamingServiceException {
+        BigInteger addressInt = new BigInteger(address.replace("0x", ""), 16);
```

Comment:
There shouldn't be an `x` in a hex address anyway, so I don't think it will make a difference. But it would make sense to validate the address with a regex in `Resolution`. I'll add some input vaildation